src/amo/imageUtils.js

```diff
@@ -1,7 +1,24 @@
 import fallbackIcon from 'amo/img/icons/default.svg';
 
-export function getAddonIconUrl(addon) {
-  return addon ? addon.icon_url : fallbackIcon;
+// returns icon url from icons objects,
+// or returns icon_url value if icons does't exists
+//
+// fallback icon will be returned if the addon has no icon.
+//
+// iconSize values are 32 | 64 | 128
+export function getAddonIconUrl(addon, iconSize) {
+  const defaultIconSize = 64;
+
+  const addonIconsExists = addon?.icons;
+
+  if (addonIconsExists) {
+    const iconFromIcons = addon.icons?.[iconSize || defaultIconSize];
+
+    return iconFromIcons || fallbackIcon;
+  }
+
+  // autocomplete API does't return icons object, returns icon_url only
+  return addon?.icon_url || fallbackIcon;
```

Comment:
Does this work?  (not tested)
```suggestion
  const iconFromIcons = addon.icons?.[iconSize] || addon.icons?.[defaultIconSize];
  
  // autocomplete API doesn't return icons object, returns icon_url only
  return iconFromIcons || addon?.icon_url || fallbackIcon;
```

`iconFromIcons` would be undefined if the specified size didn't exist, or `defaultIconSize` didn't exist, and it would then fallback to `icon_url`, and ultimately `fallbackIcon`

Reply:
Works, I added test for return defualt icon size if requested icon size is undefined.